components/pull-to-refresh.js

```diff
@@ -0,0 +1,95 @@
+import { useRouter } from 'next/router'
+import { useState, useRef, useEffect, useCallback } from 'react'
+import styles from './pull-to-refresh.module.css'
+
+const PULL_THRESHOLD = 300
+const REFRESH_TIMEOUT = 500
+
+export default function PullToRefresh ({ children, android }) {
+  const router = useRouter()
+  const [isRefreshing, setIsRefreshing] = useState(false)
+  const [pullDistance, setPullDistance] = useState(0)
+  const [isPWA, setIsPWA] = useState(false)
+  const [isAndroid, setIsAndroid] = useState(false)
+  const touchStartY = useRef(0)
+  const touchEndY = useRef(0)
+
+  const checkPWA = () => {
+    const androidPWA = window.matchMedia('(display-mode: standalone)').matches
+    const iosPWA = window.navigator.standalone === true
+    setIsAndroid(androidPWA) // we need to know if the user is on Android to enable toggling its native PTR
+    setIsPWA(androidPWA || iosPWA)
+  }
+
+  useEffect(() => {
+    checkPWA()
+  }, [])
+
+  const handleTouchStart = useCallback((e) => {
+    // don't handle if the user is not scrolling from the top of the page, is not on a PWA or if we want Android's native PTR
+    if (!isPWA || (isAndroid && !android) || window.scrollY > 0) return
+    touchStartY.current = e.touches[0].clientY
+  }, [isPWA, isAndroid, android])
+
+  const handleTouchMove = useCallback((e) => {
+    if (touchStartY.current === 0) return
```

Comment:
Is this needed so the text doesn't get stuck in some cases? At least that's what I can tell starts happening when I remove this line. A comment in the code would be nice here.

Reply:
Yes! I thought I was going crazy because between refreshes the text would get stuck, noted thanks :P